Add support for APsystem micro-converters using ECU-R

rkokkelk opened the pull request:
Continued PR from #175 . Due to rebase on different branch new PR had to be created. Main difference is that this PR is based on the `sdk-v3` branch instead of master. 

With the discussion based on the separation of responsibility in the `apsystems` module. I'm not looking forward to duplicating basically the same code and having to maintain it in two separate locations. I understand your maintainability point of view, how about I give you direct access to the repo so that you can modify it as you see fit. So that the code is not duplicated but you can still modify for maintainability of this app.

DiedB committed:
Initial commit

DiedB committed:
Delete .DS_Store

DiedB committed:
Added support for capabilities and added README

DiedB committed:
Fixed permissions and images

DiedB committed:
Added Enphase support

DiedB committed:
Update README.md

DiedB committed:
Added SolarEdge support and fixed bugs

DiedB committed:
Update README.md

DiedB committed:
Switched to http.min, added GoodWe and Omnik support, fixed bugs

DiedB committed:
Improved inverter support

DiedB committed:
Version 1.3.1: added Fronius support and changed units

DiedB committed:
Fixed GoodWe units

DiedB committed:
Update README.md

DiedB committed:
Update app.json

DiedB committed:
1.4.0

DiedB committed:
Update app.json

DiedB committed:
Update README.md

DiedB committed:
Added new app store images

DiedB committed:
Delete large.jpg

DiedB committed:
Delete small.jpg

DiedB committed:
Fixed cron issue

DiedB committed:
New images

DiedB committed:
1.4.1: Added SAJ and updated modules

DiedB committed:
Update app.json

Commit:
1.4.2: Fixed SolarEdge cron issue

DiedB committed:
Update README for 1.4.2

DiedB committed:
Update app.json

DiedB committed:
1.4.3

DiedB committed:
Rewrite PVOutput for Apps SDK v2

DiedB committed:
Converted Omnik to Apps SDK v2, multi-inverter support

DiedB committed:
Add SolarEdge to 2.0.0

DiedB committed:
Add Enphase, SolarMAN and Trannergy to v2

DiedB committed:
Add Sungrow, add more inverters to v2, update images

DiedB committed:
Add experimental Tigo support

DiedB committed:
Fix Tigo error reporting

DiedB committed:
Merge pull request #8 from DiedB/beta

Support Apps SDK v2, support Tigo

DiedB committed:
Upgrade version

DiedB committed:
Add Zeversolar support

DiedB committed:
Increase Zeversolar cron frequency

DiedB committed:
Fix Zeversolar units

DiedB committed:
Upgrade version

DiedB committed:
Add GoodWe support to v2

DiedB committed:
Upgrade version

DiedB committed:
Fix version format

DiedB committed:
Cleanup, add changelog

DiedB committed:
Add donation to app.json

DiedB committed:
2.3.2

Commit:
Fix pairing process for Homey v2.0

DiedB committed:
Improve error handling in pairing process

DiedB committed:
Merge branch 'alpha'

DiedB committed:
Upgrade Fronius to SDK v2

DiedB committed:
Basic implementation of SE production/consumption sensors

DiedB committed:
Add energy sensors, improve implementation

DiedB committed:
Refactor code, use settings, fix pairing error handling, remove broken devices

DiedB committed:
Add back SAJ support

DiedB committed:
Merge pull request #46 from DiedB/feature/solaredge-sensors

3.0.0

DiedB committed:
Fix SAJ assets

DiedB committed:
Update README.md

DiedB committed:
Fix pairing bug, fix SAJ

DiedB committed:
Fix pairing bug, fix SAJ production check

DiedB committed:
Add SAJ ID generation

DiedB committed:
Merge branch 'beta' of https://github.com/DiedB/SolarPanels into beta

DiedB committed:
Add back SolarMAN support

DiedB committed:
Fix Fronius, add SAJ voltage support

DiedB committed:
Add Omnik support

DiedB committed:
Merge pull request #50 from DiedB/beta

v3.0.1

DiedB committed:
3.0.2

DiedB committed:
Merge branch 'master' of https://github.com/DiedB/SolarPanels into beta

DiedB committed:
Merge pull request #52 from DiedB/beta

v3.0.2

DiedB committed:
Change and improve Enphase production check

DiedB committed:
Upgrade version

DiedB committed:
Ginlong implementation

DiedB committed:
Improve Omnik, fix Enphase bug

DiedB committed:
Upgrade version and README

DiedB committed:
Implement Ginlong

DiedB committed:
Merge pull request #55 from DiedB/ginlong

Ginlong implementation

DiedB committed:
Upgrade version

DiedB committed:
Fix Enphase bug

DiedB committed:
Fix more Enphase bugs

DiedB committed:
3.0.8

DiedB committed:
v3.0.9

DiedB committed:
Add license

pro-sumer committed:
#14 Support for Growatt inverters

pro-sumer committed:
#14 Review rework: use indent size of 4 spaces (and add .eslintrc config to verify)

pro-sumer committed:
#14 Review rework: remove link from developer name

pro-sumer committed:
#14 Review rework: prevent the user to go device selection without entering username/password

pro-sumer committed:
#14 Review rework: set username and password in 1 statement

pro-sumer committed:
#14 Review rework: use `await` to be consistent (code copied from code for other drivers)

pro-sumer committed:
#14 Review rework: defensive programming for empty plant names (though not sure whether this can happen)

pro-sumer committed:
#14 Work with plants instead of inverters (and use regular API to do so)

pro-sumer committed:
#14 Refactor driver using Omnik as an example

pro-sumer committed:
#14 Use class for API (as Omnik does) and verify username/password when changed in settings

pro-sumer committed:
#14 Properly support multiple plants and multiple inverters

DiedB committed:
Merge pull request #60 from pro-sumer/growatt

#14 Support for Growatt inverters

DiedB committed:
Prepare 3.1.0 release

DiedB committed:
Merge pull request #64 from DiedB/alpha

3.1.0

DiedB committed:
3.1.1

pro-sumer committed:
Fixes #65 by replacing only the first `0` by a `c` in every pair of 2 digits in the md5 hash of the user's password

DiedB committed:
Merge pull request #66 from pro-sumer/master

Fixes #65 by replacing only the first `0` by a `c` in every digits pair

DiedB committed:
3.1.1

DiedB committed:
3.1.3

DiedB committed:
Add GoodWe driver

DiedB committed:
Add GoodWe driver

DiedB committed:
First working version of GoodWe driver

DiedB committed:
Add v3.0 migration

DiedB committed:
Keep old capabilities, temporary fix for 404

DiedB committed:
Force singular selection during GoodWe pairing

DiedB committed:
Prepare v4.0.2

DiedB committed:
Merge branch 'beta' into feature/goodwe-sems

DiedB committed:
Fix GoodWe token refresh race condition

DiedB committed:
Fix GoodWe inverters

DiedB committed:
Enable Homey Compose

DiedB committed:
Remove node_modules

DiedB committed:
Move inverter class, add checkDelay to GoodWe pairing process

DiedB committed:
Finalize GoodWe checkDelay implementation

DiedB committed:
Bumps version to v4.0.5

Changelog: Improve support for GoodWe systems with multiple inverters

DiedB committed:
Fix .gitignore, rename LICENSE

DiedB committed:
Bumps version to v4.0.6

Changelog: Fix Growatt bug

DiedB committed:
Fix Growatt bug

DiedB committed:
Merge pull request #95 from DiedB/feature/goodwe-sems

Add GoodWe SEMS support

DiedB committed:
Bumps version to v4.0.8

Changelog: Update dependencies

DiedB committed:
Bumps version to v4.0.9

Changelog: Fix Growatt bug

DiedB committed:
Fix Growatt bugs

MaxvandeLaar committed:
Added Trannergy driver

MaxvandeLaar committed:
Added IntelliJ IDEA to .gitignore

MaxvandeLaar committed:
Added Max van de Laar as contributor
Version 4.1.0

DiedB committed:
Update app.json

DiedB committed:
Update package.json

DiedB committed:
Add storage support for enabled GoodWe systems

DiedB committed:
Bumps version to v4.0.10

Changelog: Add battery sensor for storage-enabled GoodWe systems

DiedB committed:
Dynamically add battery capability

DiedB committed:
Bumps version to v4.0.11

Changelog: Add battery sensor for storage-enabled GoodWe systems

DiedB committed:
Merge pull request #100 from DiedB/feature/goodwe-storage

Add storage support for GoodWe systems

DiedB committed:
Merge branch 'feature/trannergy' of https://github.com/MaxvandeLaar/Homey-SolarPanels into MaxvandeLaar-feature/trannergy

DiedB committed:
Bumps version to v4.0.12

Changelog: Add Trannergy support

DiedB committed:
Merge branch 'MaxvandeLaar-feature/trannergy'

DiedB committed:
Bumps version to v4.0.13

Changelog: Fix GoodWe API connectivity

DiedB committed:
4.0.13: Fix GoodWe, update README

DiedB committed:
Fix issue reporting URL

mikedebock committed:
implemented SolarEdge temperature measurement

mikedebock committed:
implemented list_devices view in SolarEdge pairing

DiedB committed:
Merge pull request #109 from mikedebock/solaredge-inverter-temperature

Add SolarEdge temperature, improve pairing

DiedB committed:
Bumps version to v4.0.15

Changelog: Implement flow triggers for SolarEdge inverter temperature

DiedB committed:
Add SolarEdge inverter temperature flow support

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

DiedB committed:
Add Enphase Envoy support

DiedB committed:
Update mDNS query

DiedB committed:
Bumps version to v4.1.1

Changelog: Implement Enphase Envoy (local)

DiedB committed:
Fix Envoy bugs

DiedB committed:
Fix Enphase currentEnergy null reference

DiedB committed:
Fix more Enphase bugs

DiedB committed:
Bumps version to v4.1.5

Changelog: Implement Enphase Envoy (local)

DiedB committed:
Add Solax Cloud support

DiedB committed:
Implement Kostal support

DiedB committed:
Bump version to v4.3.0

Changelog: Add support for Kostal inverters

DiedB committed:
Improve Kostal error and session handling

dependabot[bot] committed:
Bump node-fetch from 2.6.0 to 2.6.1

Bumps [node-fetch](https://github.com/bitinn/node-fetch) from 2.6.0 to 2.6.1.
- [Release notes](https://github.com/bitinn/node-fetch/releases)
- [Changelog](https://github.com/node-fetch/node-fetch/blob/master/docs/CHANGELOG.md)
- [Commits](https://github.com/bitinn/node-fetch/compare/v2.6.0...v2.6.1)

Signed-off-by: dependabot[bot] <support@github.com>

pro-sumer committed:
Use Growatt's "newTwo" API

DiedB committed:
Add source code URL to app.json

DiedB committed:
Bump version to v4.3.1

Changelog: Bump minimum Homey version, add source code URL

DiedB committed:
Merge pull request #122 from pro-sumer/master

Use Growatt's "newTwo" API

DiedB committed:
Merge branch 'fix/new-growatt-api'

DiedB committed:
Bump version to v4.3.2

Changelog: Fix Growatt API endpoint

DiedB committed:
Merge pull request #120 from DiedB/dependabot/npm_and_yarn/node-fetch-2.6.1

Bump node-fetch from 2.6.0 to 2.6.1

DiedB committed:
Merge pull request #111 from DiedB/dependabot/npm_and_yarn/lodash-4.17.19

Bump lodash from 4.17.15 to 4.17.19

DiedB committed:
Bump version to v4.3.3

Changelog: Hide consumption capability for non-supported SolarEdge devices

DiedB committed:
v4.3.4

DiedB committed:
Bump version to v4.4.0

Changelog: Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Bump version to v4.4.1

Changelog: Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Add homeyCommunityTopicId

DiedB committed:
Bump version to v4.4.3

Changelog: Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Bump version to v4.4.4

Changelog: Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Bump version to v4.4.5

Changelog: Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Bump version to v4.4.6

Changelog: Add local Omnik support, deprecate Omnik Portal connection

DiedB committed:
Add inverter temperature to local Omnik driver

DiedB committed:
Bump version to v4.5.0

Changelog: Support Huawei SUN2000L inverters

RobinBol committed:
fix: getDriver id

DiedB committed:
Add (experimental) Huawei SUN2000L support

DiedB committed:
Merge pull request #149 from RobinBol/patch-1

fix: getDriver id

DiedB committed:
Bump version to v4.5.1

Changelog: Support Huawei SUN2000L inverters

drieshooghe committed:
add esolar wifi D module driver

drieshooghe committed:
update esolar assets

drieshooghe committed:
use inverter serial number as driver id

drieshooghe committed:
adjust driver name

drieshooghe committed:
don't set the current power to 0 if the inverter is offline

drieshooghe committed:
update app.json

DiedB committed:
Bump version, move app.json to .homeycompose directory

drieshooghe committed:
update contributors

DiedB committed:
Bump version to v4.6.1

Changelog: Add support for SAJ eSolar WiFi modules

drieshooghe committed:
set device as unavailable when the status page is unreachable

DiedB committed:
Bump version to v4.6.2

Changelog: Add support for SAJ eSolar WiFi modules

DiedB committed:
Merge pull request #164 from drieshooghe/esolar-wifi-d-module

Support for eSolar Wifi D module

DiedB committed:
Create sentry-release.yml

DiedB committed:
Update sentry-release.yml

DiedB committed:
Update sentry-release.yml

rkokkelk committed:
feat: initial version of APsystems integration

rkokkelk committed:
feat: initial version of APsystems integration

rkokkelk committed:
chore[packages]: add apsystems dependency

rkokkelk committed:
chore[app]: add APsystems compose info

rkokkelk committed:
chore[app]: add APsystems compose info

rkokkelk committed:
fix[ECUR]: modify for node module v0.0.2

rkokkelk committed:
fix[ECUR]: modify for node module v0.0.2

rkokkelk committed:
chore[package]: bump apsystems to version 0.1.0

rkokkelk committed:
chore[package]: bump apsystems to version 0.1.0

rkokkelk committed:
chore[apsystems]: remove old timekeeping code

Due too unknown reasons, the cronjob timekeeping was not working
correclty. Therefore too many request were send to the device which
might crash it, forcing a powerreset.

To prevent this, code was added to ensure only 1 request is send every
minute. Cronjob timekeeping seemed to work again, so old can be removed.

rkokkelk committed:
chore[apsystems]: remove old timekeeping code

Due too unknown reasons, the cronjob timekeeping was not working
correclty. Therefore too many request were send to the device which
might crash it, forcing a powerreset.

To prevent this, code was added to ensure only 1 request is send every
minute. Cronjob timekeeping seemed to work again, so old can be removed.

rkokkelk committed:
chore[apsystems]: update assets with logo

rkokkelk committed:
chore[apsystems]: update assets with logo

rkokkelk committed:
fix[apsystem]: missing error handling

rkokkelk committed:
fix[apsystem]: missing error handling

rkokkelk committed:
docs[app]: add myself to contributing developers

rkokkelk committed:
docs[app]: add myself to contributing developers

rkokkelk committed:
chore[apsystems]: change to TypeScript

rkokkelk committed:
chore[apsystems]: change to TypeScript

rkokkelk committed:
chore[apsystems]: change driver.compose.json for sdkv3

rkokkelk committed:
chore[apsystems]: change driver.compose.json for sdkv3

DiedB commented:
Are you sure this compiles? The TypeScript compiler seems to find quite a few issues (see 'files changed', TypeScript errors are embedded in there).

rkokkelk committed:
feat[apsystems]: add types.ts definition file

rkokkelk committed:
feat[apsystems]: add types.ts definition file

rkokkelk committed:
fix[apsystems]: TypeScript errors

rkokkelk committed:
fix[apsystems]: TypeScript errors

rkokkelk committed:
feat[apsystems]: add alltypes.d.ts for

APsystems is not defined in Typescript and therefore does not contain a
type specification. In order for TS to compile properly `alltypes.d.ts`
is defined to define apsystems module.

rkokkelk committed:
feat[apsystems]: add alltypes.d.ts for

APsystems is not defined in Typescript and therefore does not contain a
type specification. In order for TS to compile properly `alltypes.d.ts`
is defined to define apsystems module.

rkokkelk commented:
There were indeed some issues which should now be resolved. Some issues mentioned by `tsc` which are shown in 'files changed' were due to typescript not detecting inheritance. I've fixed those by running:

`npm i --save-dev @types/node`

rkokkelk commented:
There were indeed some issues which should now be resolved. Some issues mentioned by `tsc` which are shown in 'files changed' were due to typescript not detecting inheritance. I've fixed those by running:

`npm i --save-dev @types/node`

DiedB committed:
Bump version to v4.6.3

Changelog: Deprecate several drivers

DiedB committed:
Driver deprecation

DiedB committed:
Create FUNDING.yml

rkokkelk committed:
Merge branch 'main' of github.com:DiedB/Homey-SolarPanels

rkokkelk committed:
chore[packages]: add apsystems dependency

rkokkelk committed:
Merge branch 'apsystems_ecur' of github.com:rkokkelk/Homey-SolarPanels into apsystems_ecur

rkokkelk committed:
merge: main

rkokkelk committed:
fix(APsystems): class reference
